projects/gateway2/translator/httproute/gateway_http_route_translator.go

```diff
@@ -372,20 +365,30 @@ func setRouteAction(
 			}
 		}
 
+		obj, err := gwroute.GetBackendForRef(backendRef.BackendObjectReference)
 		fromPlugin := false
-		for _, bp := range pluginRegistry.GetBackendPlugins() {
-			if dest, ok := bp.ApplyBackendPlugin(obj, backendRef.BackendObjectReference); ok {
-				fromPlugin = true
-				weightedDestinations = append(weightedDestinations, &v1.WeightedDestination{
-					Destination: dest,
-					Weight:      weight,
-				})
-				break
+		if err == nil {
+			for _, bp := range pluginRegistry.GetBackendPlugins() {
+				if dest, ok := bp.ApplyBackendPlugin(obj, backendRef.BackendObjectReference); ok {
+					fromPlugin = true
+					weightedDestinations = append(weightedDestinations, &v1.WeightedDestination{
+						Destination: dest,
+						Weight:      weight,
+					})
+					break
+				}
+			}
+			// TODO break out a buildDestination func to avoid this awkwardness
+			if fromPlugin {
```

Comment:
IMO using `fromPlugin` adds unnecessary complexity to the code. Here's an example of removing this bool and implementing your TODO:

```go
func applyBackendPlugins(obj interface{}, backendRef BackendObjectReference) (*DestinationType, bool) {
    for _, bp := range pluginRegistry.GetBackendPlugins() {
        if dest, ok := bp.ApplyBackendPlugin(obj, backendRef); ok {
            return dest, true
        }
    }
    return nil, false
}

obj, err := gwroute.GetBackendForRef(backendRef.BackendObjectReference)
if err == nil {
    if dest, ok := applyBackendPlugins(obj, backendRef.BackendObjectReference); ok {
        weightedDestinations = append(weightedDestinations, &v1.WeightedDestination{
            Destination: dest,
            Weight:      weight,
        })
        continue
    }
}
```

Reply:
done, thanks!